File: frontend/src/types/announcement.ts
export type AnnouncementMode = 'banner' | 'modal' | 'both'

export interface AnnouncementInput {
  title: string
  content: string
  display_mode: AnnouncementMode
  dismissible: boolean
  scrolling: boolean
  is_published: boolean
}

export interface Announcement extends AnnouncementInput {
  id: string
  revision: string
  created_at: string
  updated_at: string
}

File: internal/domain/announcement.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Announcement struct {
	ID          uuid.UUID `json:"id"`
	Title       string    `json:"title"`
	Content     string    `json:"content"`
	DisplayMode string    `json:"display_mode"`
	Dismissible bool      `json:"dismissible"`
	Scrolling   bool      `json:"scrolling"`
	IsPublished bool      `json:"is_published"`
	Revision    uuid.UUID `json:"revision"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}
